source/composer/midi_stream.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Carla on 08/05/2021.
+//
+
+#ifndef COMPOSER_MIDI_STREAM_H
+#define COMPOSER_MIDI_STREAM_H
+
+#include <MidiEventList.h>
+#include <MidiMessage.h>
+#include <utility>
+#include "midi_processor.h"
+
+// The midi_stream class uses the midi_processor class to process a smf::MidiEventList.
+// That way is possible to cycfi::q::midi to handle all MIDI messages to play the event list.
+
+class midi_stream {
+
+    static auto convert_message(smf::MidiMessage &event);
+    bool is_harmony();
```

Comment:
Não pode ser estático ou função independente?

Reply:
Coloquei todas as funções como independentes. É boa prática nesse caso criar um namespace?

Reply:
No caso, apenas o namespace do composer é suficiente. Normalmente, só precisamos de outro namespace mais interno se for (i) um conjunto muito grande de funções (ii) com propósito específico (iii) diferente do da biblioteca e (iv) que pode gerar conflito. Um namespace interno muito comum é o namespace `detail`, para colocar funções que devem ser escondidas do usuário.